content/articles/task-tracker-app-with-react.js-using-hooks/index.md

```diff
@@ -0,0 +1,226 @@
+
+## TodoForm
+`TodoForm` will be responsible for rendering the form necessary to add a to-do to the list. Using the short form, which is import react and export `TodoForm`, define a function called `TodoForm()`.
+
+In this `TodoForm()` function, go ahead and return a form element with an input and a button inside of it. Inside the `TodoForm`, you need to import `useState` because we need to define some state to keep track of input from the user.
+
+Inside the `TodoForm()` function, you need to define a state called `todo`, `setTodo()` as the function initialize as an object with three properties, ID which is a string, task which is also a string, and completed which is a boolean. Completed will be used to track whether or not the `Todo` is marked as completed, and the task will describe the `Todo`.
+
+We need to define a function for when a user types in input for a `Todo` to keep track of it in our state. We will create a function called `handleTaskInputChange()` that takes in an event as the parameter. This function will be responsible for updating the task property on our `Todo` object.
+
+In this function, we will call `setTodo`, pass in a new object with the old `Todo` property, and update the task property with the new value from the event target value. In this case, the event target value contains the new input text from the user.
+
+Our return statement will define the `onChange()` function to run our new `handleTaskInputChange()` function every time the event fires. In this case, `onChange` will fire every time the input value changes. Also, we set the input value to be `todo` task because that’s updated every time the `handleTaskInputChange()` is called. We also give our button a type called to submit. So now that we have some input data to work with, we need to handle the case when we want to add a new `todo` to the list. 
+
+ 
+
+In `app.js`, we will create a function called `addTodo()` that will take a new `todo` and add it to the array of `todos`. To add the to-do to the list, we will create a new to-do array by calling the `setTodos` function and then passing in an array with the new `todo`, add it to the beginning, and the old `todo` array spread over it.
+
+We will pass our new `addTodo()` function to the to-do form component as a prop; now, in our `todo` form component, we will destructure the `addtodo()` function from the props parameter, which in this case is the first one. 
+
+When the user submits the form, we want to then add the forms `todo` from the state to the list of to-dos; in other to do this, we are going to create a `handleSubmit()` function which also takes in an event from the DOM, here we going to call the `preventDefault()` function from preventing default browser form submit functionality then we are going to write an if statement that only gets called if the `todos` task is not empty by calling the `trim()` function on our `todo.task.trim` which will remove white space from the string.
```

Comment:
This is extremely hard to read. Break it into multiple paragraphs, kindly. Unfortunately, this is not the only case I have encountered! They are multiple which you should sort out when you read the article again and again!

In other cases, I read and get lost on the concept you're trying to put across. For example, `We will pass our new `addTodo()` function to the to-do form component as a prop; now, in our `todo` form component, we will destructure the `addtodo()` function from the props parameter, which in this case is the first one. `,, I find this sentence very hard to read.

Re-read your article and ping me when the article is ready to publish. Unfortunately, it's still quite far away!